tests/core_tests/test_client.py

```diff
@@ -0,0 +1,87 @@
+import asyncio
+
+from mock import MagicMock, patch
+
+from cloudbot.client import Client
+
+
+class Bot(MagicMock):
+    loop = asyncio.get_event_loop()
+
+
+class TestClient(Client):  # pylint: disable=abstract-method
+    _type = 'TestClient'
+    _connected = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.active = True
+
+    @property
+    def connected(self):
+        return self._connected
+
+    async def connect(self, timeout=None):
+        self._connected = True
+
+
+class FailingTestClient(TestClient):  # pylint: disable=abstract-method
+    def __init__(self, *args, fail_count=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fail_count = fail_count
+
+    async def connect(self, timeout=None):
+        if self.fail_count is not None and self.fail_count > 0:
+            self.fail_count -= 1
+            raise ValueError("This is a test")
+
+
+def test_client_no_config():
+    client = TestClient(
+        Bot(), 'foo', 'foobot', channels=['#foo']
+    )
+    assert client.config.get('a') is None
+
+
+def test_client():
+    client = TestClient(
+        Bot(), 'foo', 'foobot', channels=['#foo'], config={'name': 'foo'}
+    )
+
+    assert client.channels == ['#foo']
+    assert client.config['name'] == 'foo'
+    assert client.type == 'TestClient'
+
+    assert client.active is True
+    client.active = False
```

Comment:
Do you really need to check that an assignment you just made stuck?

Reply:
This isn't a pure value set, it's a property with a getter/setter, this is testing the getter/setter